splash-reward-distributor/src/engine/mod.rs

```diff
@@ -236,18 +277,25 @@ where
             .chain(vec![QueueCmd::DowngradeClocks(block_slot)])
             .collect(),
     };
-    queue.batch_execute(commands).await;
+    queue.clone().batch_execute(commands).await;
     tx.commit();
     ControlFlow::Continue(())
 }
 
-async fn process_tasks<GaugeId, Bearer, StateId, Q, E>(queue: Q, mut executor: E) -> ControlFlow<(), ()>
+async fn process_tasks<GaugeId, StateId, Q, E>(
+    queue: Q,
+    mut executor: E,
+    dropped_tx_hash: Option<TransactionHash>,
+) -> ControlFlow<(), ()>
 where
     Q: TaskQueue<TaskId, Task<GaugeId, StateId>> + Clone,
     E: BatchExecutor<TaskId, Task<GaugeId, StateId>, TransactionHash, ExecutorError>,
 {
     let mut invalid_tasks = vec![];
     let mut stream = queue.clone().pending_stream();
+    if let Some(tx_hash) = dropped_tx_hash {
+        reschedule_tasks_from_dropped_tx(tx_hash, queue.clone()).await;
+    }
```

Comment:
What's the point of calling it from inside process_tasks? You can use it separately in poll() method using self.block_on()